src/main/java/org/sonar/plugins/jacoco/XmlReportParser.java

```diff
@@ -42,8 +43,9 @@ public XmlReportParser(Path xmlReportPath) {
     XMLInputFactory factory = XMLInputFactory.newInstance();
     factory.setProperty(XMLInputFactory.IS_NAMESPACE_AWARE, false);
     factory.setProperty(XMLInputFactory.SUPPORT_DTD, false);
-    try {
-      XMLStreamReader parser = factory.createXMLStreamReader(Files.newBufferedReader(xmlReportPath, StandardCharsets.UTF_8));
+
+    try (Reader reader = Files.newBufferedReader(xmlReportPath, StandardCharsets.UTF_8)) {
+      XMLStreamReader parser = factory.createXMLStreamReader(reader);
```

Comment:
As discussed, we should make sure this one is closed when we don't reach the end of the document.